src/System.Data.Common/src/System/Data/Common/DbDataReaderExtensions.netcoreapp.cs

```diff
@@ -0,0 +1,181 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Data.Common
+{
+    public static partial class DbDataReaderExtensions
```

Comment:
I assume there should be documentation for the new APIs. @danmosemsft @karelz API docs get written by devs in corefx, right? If not, what's the process?

Reply:
All public API should have XML doc comments in the implementation. At present, the process for getting that to MSDN (verbatim) is rather manual. @ahsonkhan can share details. But high order bit is to make sure there are XML doc comments.

Reply:
Thanks @danmosemsft. @TheBlueSky Can you add doc comments for the new APIs please?

Reply:
@ajcvickers, can you point me to some API that are documented "the right" way, so that I do the same? The counterparts for these APIs (the `int` overloads) do not have any documentation.

Reply:
@TheBlueSky Something like this should be fine:
```C#
/// <summary>
///     Returns the boolean value from the given column in the current row.
/// </summary>
/// <param name="reader"> The reader from which to read the value. </param>
/// <param name="name"> The name of the column from which to read the value. </param>
/// <returns> The value. </returns>
public static bool GetBoolean(this DbDataReader reader, string name)
{
    AssertNotNull(reader);
    return reader.GetBoolean(reader.GetOrdinal(name));
}
```
As long as it's me signing off, I won't be too picky. :-)

Reply:
Thanks, @ajcvickers.

I'll spare sometime tomorrow or the day after to do it, if it can wait.

Reply:
@danmosemsft the XML docs are convenience which I don't think we need to enforce.
Given that there won't be any synchronization with docs after the (potential) initial, I would leave the decision to developers.

That said, we should have information about how to add docs into docs repo available soon - @carlossanlop cc @mairaw

Reply:
Correct. We can load the XML doc comments if there are some, but it's a manual process. And I'll start working on some API writing guidelines soon - the part about how we expect a summary to be, or parameters, etc.

Reply:
@karelz Oh wow! Didn't realize that you guys are still having to deal with XML docs not co-located with the source. I'm sorry. :trollface:

Reply:
Yeah, it is sadly by design in CoreFX - our code just ships in too many ways on too many platforms and OOBs, etc. :(

Reply:
Since XML docs are missing in other parts of System.Data.Common (and specifically the original int counterparts to the methods being added), is it OK if we go ahead and merge this? I can submit some documentation separately if that's desirable.

(just trying to make sure this goes through ASAP)

Reply:
@roji Yeah, that seems fine. (I'm just surprised.)

Reply:
Me too...!

Reply:
Does resolving this discussion mean no need to add XML docs to the methods? What about if I'm going to change the PR anyway; e.g., to move the extension methods to another namespace?

Reply:
@roji, sorry, I unresolved this because of my question above. Could you please let me know the decision here?

Reply:
@TheBlueSky the decision here is to not add XML docs at the moment. We may do that later.